### PR TITLE
Clean up exit 1 errors for blobfuse2

### PR DESCRIPTION
**Pull request by gapra-msft**
- Resolves https://github.com/Azure/azure-storage-fuse/issues/892
- make error strings consistent
- remove duplicate error dumps on console

### CODE REVIEW

File: cmd/doc.go [hunk 1]
```diff
@@ -51,30 +52,31 @@ var docCmd = &cobra.Command{
 	Hidden: true,
 	Short:  "Generates documentation for the tool in Markdown format",
 	Long:   "Generates documentation for the tool in Markdown format, and stores them in the designated location",
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (vibhansa-msft) [inline cmd/doc.go:55]: How are we supressing help string display on console if we return back with error ?

File: cmd/doc.go [hunk 2]
```diff
@@ -51,30 +52,31 @@ var docCmd = &cobra.Command{
...
 		// verify the output location
 		f, err := os.Stat(docCmdInput.outputLocation)
 		if err != nil && os.IsNotExist(err) {
 			// create the output location if it does not exist yet
 			if err = os.MkdirAll(docCmdInput.outputLocation, os.ModePerm); err != nil {
-				fmt.Printf("Unable to create output location due to error: " + err.Error())
-				os.Exit(1)
+				fmt.Printf("doc: Unable to create output location due to error: " + err.Error())
+				return errors.New("doc: Unable to create output location due to error: " + err.Error())
```
Comment (vibhansa-msft) [inline cmd/doc.go:62]: formattting of "doc:" is not consistent in some commands there is " : " and in some its ": ", let make them all the same.

**Review by vibhansa-msft — APPROVED** [review]

File: cmd/health-monitor.go [hunk 3]
```diff
@@ -73,63 +73,64 @@ var healthMonCmd = &cobra.Command{
 	RunE: func(_ *cobra.Command, _ []string) error {
 		resetMonitorOptions()
 
-		err := validateMonOptions()
+		err := validateHMonOptions()
 		if err != nil {
-			log.Err("health-monitor: [%v]", err)
```
Comment (souravgupta-msft) [inline cmd/health-monitor.go:78]: We should keep this log

File: cmd/health-monitor.go [hunk 4]
```diff
@@ -73,63 +73,64 @@ var healthMonCmd = &cobra.Command{
...
 	errMsg := ""
 
 	if len(pid) == 0 {
-		errMsg = "Pid of blobfuse2 process not given\n"
+		errMsg = "Pid of blobfuse2 process not given. "
 	}
 
 	if len(configFile) == 0 {
-		errMsg += "Config file not given\n"
+		errMsg += "Config file not given. "
```
Comment (souravgupta-msft) [inline cmd/health-monitor.go:130]:
```suggestion
		errMsg += "Config file not given."
```

File: cmd/mount_all.go [hunk 5]
```diff
@@ -68,37 +68,44 @@ var mountAllCmd = &cobra.Command{
...
 		return nil, cobra.ShellCompDirectiveDefault
 	},
 }
 
-func processCommand() {
-	parseConfig()
+func processCommand() error {
+	err := parseConfig()
+	if err != nil {
+		return fmt.Errorf("failed to parse config")
```
Comment (souravgupta-msft) [inline cmd/mount_all.go:92]: Add err also in the error string. Or we can keep it same as it is in mount.go where we are directly returning err.

File: cmd/mountgen1.go [hunk 6]
```diff
@@ -223,8 +221,7 @@ func generateAdlsGenOneJson() error {
 
 	err = os.WriteFile(gen1ConfigFilePath, jsonData, 0777)
 	if err != nil {
-		log.Err("mountgen1::MountRustFuse : Unable to write to adlsgen1fuse.json")
```
Comment (souravgupta-msft) [inline cmd/mountgen1.go:226]: Keep this log

File: cmd/mount.go [hunk 7]
```diff
@@ -248,13 +250,15 @@ var mountCmd = &cobra.Command{
 		}
 
 		if configFileExists {
-			parseConfig()
+			err := parseConfig()
+			if err != nil {
+				return err
```
Comment (souravgupta-msft) [inline cmd/mount.go:255]: We should keep this consistent in all the places where parseConfig() method is called

File: cmd/mount.go [hunk 8]
```diff
@@ -391,9 +392,7 @@ var mountCmd = &cobra.Command{
 		log.Crit("Logging level set to : %s", logLevel.String())
 		pipeline, err = internal.NewPipeline(options.Components, !daemon.WasReborn())
 		if err != nil {
-			log.Err("Mount: error initializing new pipeline [%v]", err)
```
Comment (souravgupta-msft) [inline cmd/mount.go:394]: Keep this log

File: cmd/mount.go [hunk 9]
```diff
@@ -408,26 +407,29 @@ var mountCmd = &cobra.Command{
 			daemon.SetSigHandler(sigusrHandler(pipeline, ctx), syscall.SIGUSR1, syscall.SIGUSR2)
 			child, err := dmnCtx.Reborn()
 			if err != nil {
-				log.Err("Mount: error daemonizing application [%v]", err)
```
Comment (souravgupta-msft) [inline cmd/mount.go:411]: Keep this log

File: cmd/health-monitor.go (hunk 3 above)
Comment (gapra-msft) [reply cmd/health-monitor.go:78]: fixed

File: cmd/root_test.go [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+func (suite *rootCmdSuite) SetupTest() {
+	suite.assert = assert.New(suite.T())
+	err := log.SetDefaultLogger("silent", common.LogConfig{Level: common.ELogLevel.LOG_DEBUG()})
+	if err != nil {
+		panic("Unable to set silent logger as default.")
+	}
+}
+
+func (suite *rootCmdSuite) cleanupTest() {
+	resetCLIFlags(*generateConfigCmd)
```
Comment (souravgupta-msft) [inline cmd/root_test.go:60]:
```suggestion
	resetCLIFlags(*rootCmd)
```

**Review by souravgupta-msft — APPROVED** [review]

**Review by vibhansa-msft — APPROVED** [review]

**Review by tasherif-msft — APPROVED** [review]

**Pull request merged**